### ISSUE TITLE
so_rcvbuf is quadrupled instead of doubled

### ISSUE THREAD

**Issue by redward** [multipath-tcp/mptcp#36]
calling setsockopt on 'so_rcvbuf' before listen(), it results in a buffer multiplied by 4 instead by 2. 
found with packetdrill_mptcp, example in :
https://github.com/ArnaudSchils/packetdrill_mptcp/blob/master/packetdrill/gtests/net/packetdrill/examples/tests/sock_headers/so_rcvbuf-sndbuf_set_before_listen.pkt

**Comment by cpaasch** [comment]
Fixed, but pushed only to mptcp_trunk, because it is not a crash and the change is quite a big change to the initial value of the receive/send-buffer.

**Comment by dykelin** [comment]
when this bug closed, make this change:

”-    if (tp->mpc) {
”+    if (tp->mpc)
        mptcp_init_buffer_space(sk);
“-        mptcp_update_sndbuf(tp->mpcb);
”-    }
 }

but I download the 0.88.12 source code last week,found that this change was reverd,Which one is the ultimate right?

**Comment by cpaasch** [comment]
Look at my comment https://github.com/multipath-tcp/mptcp/issues/36#issuecomment-41652754

```
Fixed, but pushed only to mptcp_trunk, because it is not a crash and the change is quite a big change to the initial value of the receive/send-buffer.
```

The fix has not been backported to v0.88.x, because it does not fix a crash and is rather a big change.

**Comment by dykelin** [comment]
OK,thanks!
